app/models/task.py

```diff
@@ -1,5 +1,47 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from typing import Optional
+from datetime import datetime
+from typing import Optional
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] 
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def to_dict(self):
+        return {
+            "task": {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_complete(),
+            "goal_id": self.goal.id if self.goal else None
+            }
+        }
+
+    def to_dict_without_goal_id(self):
+        return {
+            "task": {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_complete(),
+            }
+        }
```

Comment:
Having two distinct methods duplicates 5 lines of logic across these two methods. How can you write conditional logic in `to_dict` so that you don't need `to_dict_without_goal_id` at all?

What if instead of just returning a dictionary literal, like you did in `to_dict` for `Goal`, you create a variable that references the dictionary, write some logic related to the goal id and set the attribute if necessary, and then return it?

```py
    def to_dict(self):
        task = {
            "task": {
            "id": self.id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.is_complete(),
            }
        }
        if self.goal:
                task["goal_id"] =  self.goal.id

        return task
```